### PR TITLE
Creates GET /api/articles endpoint

### PR DESCRIPTION
**Pull request by Burnlees**
Hi Northcoders,

This branch:
- creates GET /api/articles endpoint
- creates model and controller needed to send back the correct response
- creates a utility function to bulk remove a property from an array of objects with testing

thanks for reviewing my code :)

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -96,3 +97,36 @@ describe("GET /api/articles/:article_id", () => {
       });
   });
 });
+
+describe("GET /api/articles", () => {
+  it("should respond with a 200 status code when successful", () => {
+    return request(app).get("/api/articles").expect(200);
```
Comment (Harrison365) [inline __tests__/app.test.js:103]: Not needed. You are checking for 200s already below 🙂

File: __tests__/app.test.js [hunk 2]
```diff
@@ -96,3 +97,36 @@ describe("GET /api/articles/:article_id", () => {
...
+          expect(article).toMatchObject({
+            article_id: expect.any(Number),
+            title: expect.any(String),
+            topic: expect.any(String),
+            author: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            article_img_url: expect.any(String),
+            comment_count: expect.any(Number),
+          });
```
Comment (Harrison365) [inline __tests__/app.test.js:120]: Great. However, if you are using a forEach, you need to test the length of the array to avoid false positives for an empty array.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -96,3 +97,36 @@ describe("GET /api/articles/:article_id", () => {
...
+        });
+      });
+  });
+  it("should when successful respond with an array of articles sorted by date in descending order", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toBeSortedBy("created_at", { descending: true });
+      });
```
Comment (Harrison365) [inline __tests__/app.test.js:130]: Very good 👍

File: __tests__/utils.test.js [hunk 4]
```diff
@@ -102,3 +103,148 @@ describe("formatComments", () => {
...
+        body: "I find this existence challenging",
+        created_at: 1594329060000,
+        votes: 100,
+        article_img_url:
+          "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+      },
+    ];
+    const inputTwo = "body";
+    const actual = removePropertyFromObjectArray(inputOne, inputTwo);
+    expect(actual).not.toBe(inputOne)
```
Comment (Harrison365) [inline __tests__/utils.test.js:248]: Clever, but we mustn't remove the body from our articles using JavaScript, we must edit our SQL query to get all the columns except the body. Please make sure you are doing this 🙂

File: controllers/articles.controllers.js [hunk 5]
```diff
@@ -1,9 +1,20 @@
...
+const {
+  selectArticleById,
+  selectArticles,
+} = require("../models/articles.models");
+
+exports.getArticles = (req, res, next) => {
+  selectArticles().then((articlesData) => {
+    res.status(200).send({ articles: articlesData });
+  });
+};
```
Comment (Harrison365) [inline controllers/articles.controllers.js:10]: Good 👍

File: models/articles.models.js [hunk 6]
```diff
@@ -1,4 +1,23 @@
...
+      SELECT articles.*, CAST(COUNT(comments.article_id) AS INTEGER) AS comment_count
+      FROM articles
+      JOIN comments ON articles.article_id = comments.article_id
+      GROUP BY articles.article_id
+      ORDER BY articles.created_at DESC
+    `
+    )
+    .then((res) => {
+      const articleData = removePropertyFromObjectArray(res.rows, "body");
+
```
Comment (Harrison365) [inline models/articles.models.js:17]: Again, please use SQL not JS to exclude the body 🙂
